### PR TITLE
Feature/daniel/user profile

### PR DESCRIPTION
**Pull request by cimthog**
created user profile model and routes and a validation function

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -15,7 +16,7 @@ const handle = nextApp.getRequestHandler();
 // Configure DB
 const db = require('../config/keys').mongoURI;
 // Connect to MongoDB
-mongoose.connect(db);
+mongoose.connect(db, { useNewUrlParser: true });
```
Comment (Way2nnadi) [inline server/index.js:19]: what is `useNewUrlParser` doing?

File: validation/profile.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    const errors = {};
+
+    let {
+        city, country, email, firstName, lastName,
+    } = data;
+    city = !isEmpty(city) ? city : '';
+    country = !isEmpty(country) ? country : '';
+    email = !isEmpty(email) ? email : '';
+    firstName = !isEmpty(firstName) ? firstName : '';
+    lastName = !isEmpty(lastName) ? lastName : '';
```
Comment (Way2nnadi) [inline validation/profile.js]: I do not agree with this pattern of using `isEmpty` here. This should be 
```
const {
    city = '',
    country = '',
    email = '',
    firstName = '',
    lastName = '',
} = data;
```

File: routes/api/profile.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        const updatedProfile = await Profile.findByIdAndUpdate(profileId, profile, { new: true });
+        if (!updatedProfile) {
+            return res.json({
+                message: `Profile not found with id ${profileId}`,
+            });
+        }
+        return res.json(updatedProfile);
+    } catch (err) {
+        if (err.kind === 'ObjectId') {
+            return res.status(404).json({
```
Comment (Way2nnadi) [inline routes/api/profile.js]: what do you think `404` means?

File: server/index.js (hunk 1 above)
Comment (cimthog) [reply server/index.js:19]: I was getting this error

(node:4024) DeprecationWarning: current URL string parser is deprecated, and will be removed in a future version. To use the new parser, pass option { useNewUrlParser: true } to MongoClient.connect.

so i had to pass it to my MongoClient.connect

File: validation/profile.js (hunk 2 above)
Comment (cimthog) [reply validation/profile.js]: ok, fixed

File: server/index.js (hunk 1 above)
Comment (Way2nnadi) [reply server/index.js:19]: okay

**Review by Way2nnadi — DISMISSED** [review]
LGTM

**Review by Way2nnadi — APPROVED** [review]

**Comment by Way2nnadi** [comment]
this is ready to be merged, @cimthog get approval from the rest of your team

**Pull request merged**